### PR TITLE
Allow symbols as Weak{Map,Set} keys

### LINKED ISSUE #1194: Symbols can't be used as WeakMap keys
Symbols cannot be used as WeakMap keys as they are primitives. However unlike other primitives they are unforgable and unique, so they should be completely safe for use as WeakMap keys.

### PR DESCRIPTION
**Pull request by devsnek**
Fixes #1194

Right now this doesn't allow symbols from the global registry but that can change.

this replaces #1206 because my fork of ecma262 was deleted since then.... sorry for the churn.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -34329,7 +34329,10 @@ <h1>WeakMap.prototype.delete ( _key_ )</h1>
           1. If Type(_M_) is not Object, throw a *TypeError* exception.
           1. If _M_ does not have a [[WeakMapData]] internal slot, throw a *TypeError* exception.
           1. Let _entries_ be the List that is _M_.[[WeakMapData]].
-          1. If Type(_key_) is not Object, return *false*.
+          1. If Type(_key_) is Symbol, then,
+            1. For each element _e_ of the GlobalSymbolRegistry List, do,
+              1. If SameValue(_e_.[[Symbol]], _value_) is true, return *false*.
```
Comment (gibson042) [inline spec.html]: As noted in https://github.com/tc39/ecma262/issues/1194#issuecomment-393143921 , I object to this increase in complexity and cognitive burden.

**Comment by devsnek** [comment]
so will this need to be a proposal? I was considering it a spec bug since any unforgable value can be gc'd and would therefore make sense as the key of a weakmap entry.

**Comment by ljharb** [comment]
It would need to be a proposal, or a needs consensus PR, i believe.

**Comment by bterlson** [comment]
I think this should start life as a proposal, but if we'd prefer a PR we can re-open.

**Pull request closed without merging**